pkg/cache/context.go

```diff
@@ -1152,9 +1161,13 @@ func (ctx *Context) RemoveTask(appID, taskID string) {
 	app.RemoveTask(taskID)
 }
 
-func (ctx *Context) getTask(appID string, taskID string) *Task {
+func (ctx *Context) GetTask(appID string, taskID string) *Task {
 	ctx.lock.RLock()
 	defer ctx.lock.RUnlock()
+	return ctx.getTask(appID, taskID)
+}
+
+func (ctx *Context) getTask(appID, taskID string) *Task {
```

Comment:
This is not needed if you first retrieve the task using `getTask` before calling `AssumePod` in the tests. Also becomes unneeded if we handle the fail in the `UpdateAllocation`

Reply:
Done